[25.1] Do not update a user's update_time when an admin archives (and purges) a history

natefoo opened the pull request:
Fixes #17809.

Thanks to Claude I finally did TDD for the first time on this project. :stuck_out_tongue_winking_eye:

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

natefoo committed:
Do not update a user's update_time when an admin archives (and purges) a history

jmchilton commented on test/integration/test_history_archiving.py:
```diff
@@ -239,3 +243,110 @@ def _export_history_to_short_term_storage(self, history_id):
         return last_record
+
+
+class TestHistoryArchivingUserUpdateTime(PosixFileSourceSetup, IntegrationTestCase):
```
Very nice. Ask Claude if it has any reason to not just add these test methods to the existing class in this directory? The Galaxy config looks the same so I wouldn't create another test instance just for these two tests - but maybe it has a good reason.

natefoo replied:
Ah well we'd have to set the `require_admin_user` class property, not sure if we want to do that for the existing tests. Do we have a per-test-method way to do it?

jdavcs commented:
> Not sure if this 100% equivalent in SQLAlchemy, but this fixes the mypy errors for me:

Yes, it is equivalent.

natefoo committed:
Fix mypy errors in test

nsoranzo committed:
Fix mypy errors in changes to recalculate_user_disk_usage()

jmchilton committed:
Deduplicate history archive test additions

nsoranzo committed:
Just run ``execute()``

as in the other `if` branch.

Fix:

```
sqlalchemy.exc.InvalidRequestError: A transaction is already begun on this Session.
```

natefoo committed:
Update lib/galaxy/webapps/galaxy/services/histories.py

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

mvdbeek commented:
If I'm not mistaken this is the first PR that led to No space left on disk: https://github.com/galaxyproject/galaxy/actions/runs/20469545248/job/58821408376#step:13:6497
